### ISSUE TITLE
Can't use Schedulers.from() in a child of RxJavaSchedulersHook class.

### ISSUE THREAD

**Issue by rsavin** [ReactiveX/RxJava#2918]
I use RxJava on Google App Engine. GAE forces to use `com.google.appengine.api.ThreadManager` for creating new threads. So I had to implement RxJavaSchedulersHook for creating schedulers that use `ThreadManager`

I've created an extension of RxJavaSchedulersHook. It looked like this:

```java
public class MyRxThreadSchedulersHook extends RxJavaSchedulersHook {

private final Scheduler scheduler = Schedulers.from(new ScheduledThreadPoolExecutor(3, ThreadManager.backgroundThreadFactory()));

    @Override
    public Scheduler getComputationScheduler() {
        return scheduler;
    }

    @Override
    public Scheduler getIOScheduler() {
        return scheduler;
    }

    @Override
    public Scheduler getNewThreadScheduler() {
        return scheduler;
    }

}
```
But I get the following exception:
`java.lang.IllegalStateException: Another strategy was already registered: rx.plugins.RxJavaSchedulersHook`

The reason is that `Schedulers` has  no lazy initialization, so when I call `Schedulers.from()` the Schedulers instance is initialized and when I call  ` RxJavaPlugins.registerSchedulersHook();` I can't register my hook.

So I had to create an instance of `ExecutorScheduler` via reflection (It's no public) and my RxJavaSchedulersHook looks like this:

```java

public class MyRxThreadSchedulersHook extends RxJavaSchedulersHook {

    private final Executor executor = new ScheduledThreadPoolExecutor(3, ThreadManager.backgroundThreadFactory());

    private final Scheduler scheduler;
    {
        try {
            final Constructor<?> constructor = Class.forName("rx.schedulers.ExecutorScheduler").getConstructor(Executor.class);
            constructor.setAccessible(true);
            scheduler = (Scheduler) constructor.newInstance(executor);
        } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException | ClassNotFoundException e) {            
            scheduler = null;
        }
    }

... //methods implementation
```

It's bad practice to use reflection for third-party libraries, so could you implement lazy initialization for `Schedulers` or suggest another solution of my problem, please?

Thanks.

**Comment by iNoles** [comment]
Have you seen https://github.com/ReactiveX/RxJava/wiki/Plugins ?

**Comment by rsavin** [comment]
@iNoles Yes, I've seen it.
As I said, I implemented `RxJavaSchedulersHook`. But when I call `registerSchedulersHook( )` I get `IllegalStateException`. 
I described the reason of this exception in my case and my workaround. But this workaround uses reflection so it's bad solution. 
I hope, that I'll can use `Schedulers.from()` in my `RxJavaSchedulersHook` in future (as I've showed in the first peace of code).
